lightning/src/ln/channel.rs

```diff
@@ -3442,6 +3618,15 @@ impl<Signer: Sign> Channel<Signer> {
 			return Err(ChannelError::Close("Remote tried to send us a closing tx with > 21 million BTC fee".to_owned()));
 		}
 
+		if self.is_outbound() && self.last_sent_closing_fee.is_none() {
+			return Err(ChannelError::Close("Remote tried to send a closing_signed when we were supposed to propose the first one".to_owned()));
+		}
+
+		if self.channel_state & ChannelState::MonitorUpdateFailed as u32 != 0 {
+			self.pending_counterparty_closing_signed = Some(msg.clone());
+			return Ok((None, None));
+		}
+
 		let funding_redeemscript = self.get_funding_redeemscript();
 		let (mut closing_tx, used_total_fee) = self.build_closing_transaction(msg.fee_satoshis, false);
 		if used_total_fee != msg.fee_satoshis {
```

Comment:
Since we're editing this area, should the error message below be "Remote sent us a closing_signed with a fee other than the value they can claim. Fee in message: {}. Actual fee of closing tx: {}"